lib/index.js

```diff
@@ -682,6 +683,13 @@ var Client = module.exports = function(config) {
                     basic = new Buffer(this.auth.username + ":" + this.auth.password, "ascii").toString("base64");
                     headers["Authorization"] = "Basic " + basic;
                     break;
+                case "netrc":
+                    var auth = netrc()["api.github.com"];
```

Comment:
Actually that might be necessary since there is a rare instance in which an endpoint uses a custom host. repos.uploadAsset() uses "host": "uploads.github.com". Since the netrc uses machine <host> the extra param should probably be called machine.